connect/api/src/main/java/org/apache/kafka/connect/health/ConnectClusterState.java

```diff
@@ -17,29 +17,29 @@
 
 package org.apache.kafka.connect.health;
 
-import org.apache.kafka.connect.rest.ConnectRestExtension;
-
 import java.util.Collection;
 
 /**
- * Provides the ability to lookup connector metadata and its health. This is made available to
- * the {@link ConnectRestExtension} implementations. Connect Framework provides the implementation
- * for this interface.
+ * Provides the ability to lookup connector metadata and its health. This is made available to the {@link org.apache.kafka.connect.rest.ConnectRestExtension}
+ * implementations. Connect Framework provides the implementation for this interface.
  */
 public interface ConnectClusterState {
 
     /**
-     * Get the names of the connectors currently deployed in this cluster. This is a full list of
-     * connectors in the cluster gathered from the current configuration.
-     * @return collection of connector names
+     * Get the names of the connectors currently deployed in this cluster. This is a full list of connectors in the cluster gathered from
+     * the current configuration.
+     *
+     * @return collection of connector names, never {@code null}
      */
     Collection<String> connectors();
 
     /**
-     * Lookup the current health of a connector and its tasks.
+     * Lookup the current health of a connector and its tasks. This provides the current snapshot of health by querying the underlying
+     * herder. A connector returned by previous invocation of {@link #connectors()} may no longer be available and could result in
+     * {@link org.apache.kafka.connect.errors.NotFoundException}
      *
      * @param connName name of the connector
-     * @return ConnectorHealth for the provided {@param connName}
+     * @return the health of the connector for the connector name, may be {@code null}
```

Comment:
The method description in the JavaDoc says that a `NotFoundException` will be thrown if no connector with the supplied name exists when this method is called, but the `@return` description says that the method can return null. We should decide: is it better to throw an exception or return null.

If we throw an exception, we should add `@throws NotFoundException if a connector with the supplied name does not exist` to fully document the behavior. (In general, the Kafka JavaDocs are relatively poor on this front.)